File: serial_comm/HandshakeMessage.cpp
#include "HandshakeMessage.h"

#include "Arduino.h"
#include "Messages.h"
#include "Utils.h"

HandshakeMessage::HandshakeMessage() :
MessageBase(Messages::HANDSHAKE),
payload(0u)
{
  
}

Vector<unsigned char> HandshakeMessage::encode() const
{
  unsigned char buf[2u] = {(unsigned char) (payload & 0xFF), (unsigned char) (payload >> 8)};
  
  Vector<unsigned char> retVal(1);
  retVal.push_back(buf[0]);
  retVal.push_back(buf[1]);
  
  return retVal;
}

void HandshakeMessage::decode(const unsigned char* data, const uint16_t size)
{
  UNUSED(size);
  payload = 0;
  payload = data[1] << 8u | (data[0] & 0x00FF);
}

File: python/arduino_termpoint/comm_messaging/comm_serial/handshake_message.py
from ..message import Message
from .messages import Messages
import struct

class HandshakeMessage(Message):
    offset = 0
    def __init__(self):
        self.payload = 0

    def get_msg_type(self):
        return Messages.HANDSHAKE

    def encode(self):
        HandshakeMessage.offset += 1
        if HandshakeMessage.offset > 10000:
            HandshakeMessage.offset = 0
        to_send = self.payload + HandshakeMessage.offset
        return struct.pack('<h', to_send)

    def decode(self, data):
        self.payload = struct.unpack('<h', data)[0]
